extensions/spark/kyuubi-extension-spark-3-5/src/main/scala/org/apache/kyuubi/sql/watchdog/MaxScanStrategy.scala

```diff
@@ -232,6 +234,54 @@ case class MaxScanStrategy(session: SparkSession)
               logicalRelation.catalogTable)
           }
         }
+      case ScanOperation(
+            _,
+            _,
+            _,
+            relation @ DataSourceV2ScanRelation(_, _, _, _, _)) =>
+        val table = relation.relation.table
+        if (table.partitioning().nonEmpty &&
+          relation.scan.isInstanceOf[SupportsReportPartitioning]) {
+          val partitionColumnNames = table.partitioning().map(_.describe())
+          val stats = relation.computeStats()
+          lazy val scanFileSize = stats.sizeInBytes
+          lazy val scanPartitions = relation.scan.asInstanceOf[SupportsReportPartitioning]
+            .outputPartitioning()
+            .numPartitions()
```

Comment:
`numPartitions` does not seem to be the number of scan table partitions. As in iceberg implementation, it is the size of `taskGroups`.

![image](https://github.com/apache/kyuubi/assets/17894939/7d673548-778b-41ed-8a9c-756765a2f9b3)

Reply:
I think it is the task number of RDD/stage, instead of the table's partition number, does taskGroups in Iceberg means same thing?

Reply:
It's the input RDD partition number for iceberg datasource.  Maybe the value of it is equal to table's partition number, but they're not the same thing. Seems it's a bit hard to get the number of scan table partitions.